server/base/src/main/java/org/apache/accumulo/server/metadata/ServerAmpleImpl.java

```diff
@@ -65,14 +68,29 @@ public class ServerAmpleImpl extends AmpleImpl implements Ample {
 
   private final ServerContext context;
   private final ScanServerRefStore scanServerRefStore;
+  private final Supplier<ConditionalWriter> sharedMetadataWriter;
+  private final Supplier<ConditionalWriter> sharedUserWriter;
```

Comment:
Seemes like these always come from the context, so maybe we could not store them and always get them from the context when needed.  That may simplify the constructors in this class a bit.

Reply:
Yea good idea. Fixed that in b588423